src/provider/index.ts

```diff
@@ -134,6 +135,7 @@ export class Main {
                     };
                     this._eventPump.push<NotificationActionEvent>(source.uuid, event);
                 }
+                this._store.dispatch(new RemoveNotifications([action.notification]));
```

Comment:
I'm feeling nervous about all the un-awaited calls to `store.dispatch` in `index.ts`. Is this definitely the right thing to do? Are we sure that everything we want to happen before we return to the client actually does happen before we return to the client? @TheoOttahOF, any view on this?

Reply:
If we want `await` here, we want `await` on the other dispatch calls in `index.ts` right?

I'm not necessarily saying we do want to `await` these calls. I just want to make sure careful thought has been given to whether or not we want to `await` the dispatch calls.

Reply:
I don't think it should be awaited. The state gets updated which we care about, but we don't care about how that action is being handled (toast animating out), so awaiting would add more time to the dispatcher of the `ClickNotification` getting a resolved response.

Reply:
Oooh, I didn't realised animations would be wrapped up in the promise returned here. I feel like there's still an in person conversation to be had, but let's not let that delay this PR.

Sorry for the flip-flop Noyan! I suggest let's revert to no-await. Maaaaybe that's something want to change later, but seems like a thorny enough question to address separately

Reply:
I agree at this stage there is no use case and the await is not necessary since the dispatching of a close action is literally the last thing we do down the line, but don't think "awaiting would add more time to the dispatcher of the ClickNotification getting a resolved response" bit is not necessarily true again for the same reason, at the time we dispatch the closed action, both the state update and the "post-action" of click notification is handled as this is literally the last thing we do down the line.